### DISCUSSION TITLE
evaluation_of_RAG_retrieval_chunking_methods.md - how did you guys calculate precision?

### DISCUSSION THREAD

**Discussion by aidando73** [superlinked/VectorHub#533] [Discussions]
- https://github.com/superlinked/VectorHub/blob/main/docs/articles/evaluation_of_RAG_retrieval_chunking_methods.md

I'm trying to reproduce this calculation (squad dataset)

![AI Board](https://github.com/user-attachments/assets/30635052-a55d-472d-b3bd-215adf2d10e4)

But I'm getting 0.213 instead of 0.1246

My code is here: https://github.com/aidando73/hello-retrieval-evals/blob/main/squad_1.ipynb

What am I doing differently here? I'm doing:
- Using same model
- Using ScentenceSplitter same chunk size (no chunk overlap)
- Consider the result correct if it contains the answer as a substring
- Using chromadb's default kNN algorithm
- I tried all different splits (train, validation, train + validation)

**Comment by svonava-superlinked** [comment]
@hkristof03 could you check this out?

**Reply by hkristof03** [reply]
The code is open sourced, one reason to be able to reproduce it. Sorry but I don't do random requests with different code.

**Reply by svonava-superlinked** [reply]
Ah that's right, I totally forgot the code was in a separate repo!

@aidando73 could you check https://github.com/superlinked/chunking-research and see where your code might have deviated?

**Comment by aidando73** [comment]
Sweet - thank you 🙏
